internal/handlers/task_handler.go

```diff
@@ -0,0 +1,266 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/DA-Melentev/go_final_project/internal/models"
+	"github.com/DA-Melentev/go_final_project/internal/services"
+	"github.com/DA-Melentev/go_final_project/internal/utils"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type TaskHandler struct {
+	Service *services.TaskService
+}
+
+func NewTaskHandler(service *services.TaskService) *TaskHandler {
+	return &TaskHandler{Service: service}
+}
+
+func (h *TaskHandler) NextDate(w http.ResponseWriter, r *http.Request) {
+	nowStr := r.URL.Query().Get("now")
+	date := r.URL.Query().Get("date")
+	repeat := r.URL.Query().Get("repeat")
+
+	log.Printf("New nextdate request {now:%s date:%s repeat:%s}", nowStr, date, repeat)
+
+	now, err := time.Parse("20060102", nowStr)
+	if err != nil {
+		http.Error(w, "Invalid now param", http.StatusBadRequest)
+		log.Printf("Invalid now param: %v", err)
+		return
+	}
+
+	_, err = time.Parse("20060102", date)
+	if err != nil {
+		http.Error(w, "Invalid date param", http.StatusBadRequest)
+		log.Printf("Invalid date param: %v", err)
+		return
+	}
+
+	result, err := utils.NextDate(now, date, repeat)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		log.Printf("error while calculating nextdate: %v", err)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/plain")
+	fmt.Fprint(w, result)
+}
+
+func (h *TaskHandler) AddTask(w http.ResponseWriter, r *http.Request) {
+	var task models.Task
+	err := json.NewDecoder(r.Body).Decode(&task)
+	if err != nil {
+		WriteError(w, http.StatusBadRequest, err)
+		log.Printf("Error while parsing json: %v", err)
+		return
+	}
+	log.Printf("/api/task POST: %v", task)
+
+	status, err := handleTask(&task)
+	if err != nil {
+		WriteError(w, status, err)
+		log.Printf("error: %v", err)
+		return
+	}
+
+	id, err := h.Service.AddTask(task)
+	if err != nil {
+		WriteError(w, http.StatusInternalServerError, err)
+		log.Printf("error: %v", err)
+		return
+	}
+
+	WriteResponseJSON(w, http.StatusCreated, map[string]interface{}{
+		"id": id,
+	})
+}
+
+func (h *TaskHandler) PutTask(w http.ResponseWriter, r *http.Request) {
+	var task models.Task
+	err := json.NewDecoder(r.Body).Decode(&task)
+	if err != nil {
+		WriteError(w, http.StatusBadRequest, err)
+		log.Printf("Error while parsing json: %v", err)
+		return
+	}
+	log.Printf("/api/task PUT: %v", task)
+
+	status, err := handleTask(&task)
+	if err != nil {
+		WriteError(w, status, err)
+		log.Printf("error: %v", err)
```

Comment:
Не скупись на логах, сейчас, когда мало клиентов, ты можешь отследить порядок логов, но когда много клиентов будет подряд идти, то будет каша в логах. (Да там начинается история с хорошим логеров, request-id и тд, но лучше немного больше писать)

```suggestion
		log.Printf("error while handleTask for PUT: %v", err)
```